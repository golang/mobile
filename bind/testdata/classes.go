// Copyright 2014 The Go Authors.  All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package java

import (
	gopkg "Java/java"
	"Java/java/io"
	"Java/java/lang"
	"Java/java/lang/System"
	"Java/java/util/Spliterators"
	"Java/java/util/concurrent"
)

type Runnable struct {
	lang.Runnable
}

func (r *Runnable) Run(this gopkg.Runnable) {
}

type InputStream struct {
	io.InputStream
}

func (_ *InputStream) Read() (int32, error) {
	return 0, nil
}

func NewInputStream() *InputStream {
	return new(InputStream)
}

type Future struct {
	concurrent.Future
}

func (_ *Future) Get() (lang.Object, error) {
	return nil, nil
}

// Use a trailing underscore to override multiple overloaded methods.
func (_ *Future) Get_(_ int64, _ concurrent.TimeUnit) (lang.Object, error) {
	return nil, nil
}

type Object struct {
	lang.Object
}

func innerClassTypes() {
	// java.util.Spliterators.iterator use inner class types
	// for the return value as well as parameters.
	Spliterators.Iterator(nil)
}

func returnType() {
	// Implicit types (java.io.Console) should be wrapped.
	cons := System.Console()
	cons.Flush()
}
