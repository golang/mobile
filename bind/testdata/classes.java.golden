// Java class java.Future is a proxy for talking to a Go program.
//   gobind -lang=java classes
//
// File is generated by gobind. Do not edit.
package java;

import go.Seq;

public final class Future implements Seq.GoObject, java.util.concurrent.Future {
    static { Java.touch(); }
    
    private final int refnum;
    
    @Override public final int incRefnum() {
          Seq.incGoRef(refnum, this);
          return refnum;
    }
    
    Future(int refnum) { this.refnum = refnum; Seq.trackGoRef(refnum, this); }
    
    public Future() { this.refnum = __New(); Seq.trackGoRef(refnum, this); }
    
    private static native int __New();
    
    public final native java.util.concurrent.Future getFuture();
    public final native void setFuture(java.util.concurrent.Future v);
    
    @Override public native java.lang.Object get() throws java.lang.InterruptedException, java.util.concurrent.ExecutionException;
    /**
     * Use a trailing underscore to override multiple overloaded methods.
     */
    @Override public native java.lang.Object get(long p0, java.util.concurrent.TimeUnit p1) throws java.lang.InterruptedException, java.util.concurrent.ExecutionException, java.util.concurrent.TimeoutException;
}

// Java class java.InputStream is a proxy for talking to a Go program.
//   gobind -lang=java classes
//
// File is generated by gobind. Do not edit.
package java;

import go.Seq;

public final class InputStream extends java.io.InputStream implements Seq.GoObject {
    static { Java.touch(); }
    
    private final int refnum;
    
    @Override public final int incRefnum() {
          Seq.incGoRef(refnum, this);
          return refnum;
    }
    
    public InputStream() {
        super();
        this.refnum = __NewInputStream();
        Seq.trackGoRef(refnum, this);
    }
    
    private static native int __NewInputStream();
    
    public final native java.io.InputStream getInputStream();
    public final native void setInputStream(java.io.InputStream v);
    
    @Override public native int read() throws java.io.IOException;
}

// Java class java.Object is a proxy for talking to a Go program.
//   gobind -lang=java classes
//
// File is generated by gobind. Do not edit.
package java;

import go.Seq;

public final class Object extends java.lang.Object implements Seq.GoObject {
    static { Java.touch(); }
    
    private final int refnum;
    
    @Override public final int incRefnum() {
          Seq.incGoRef(refnum, this);
          return refnum;
    }
    
    Object(int refnum) { this.refnum = refnum; Seq.trackGoRef(refnum, this); }
    
    public Object() { this.refnum = __New(); Seq.trackGoRef(refnum, this); }
    
    private static native int __New();
    
    public final native java.lang.Object getObject();
    public final native void setObject(java.lang.Object v);
    
}

// Java class java.Runnable is a proxy for talking to a Go program.
//   gobind -lang=java classes
//
// File is generated by gobind. Do not edit.
package java;

import go.Seq;

public final class Runnable implements Seq.GoObject, java.lang.Runnable {
    static { Java.touch(); }
    
    private final int refnum;
    
    @Override public final int incRefnum() {
          Seq.incGoRef(refnum, this);
          return refnum;
    }
    
    Runnable(int refnum) { this.refnum = refnum; Seq.trackGoRef(refnum, this); }
    
    public Runnable() { this.refnum = __New(); Seq.trackGoRef(refnum, this); }
    
    private static native int __New();
    
    public final native java.lang.Runnable getRunnable();
    public final native void setRunnable(java.lang.Runnable v);
    
    @Override public native void run();
}

// Java class java.Java is a proxy for talking to a Go program.
//   gobind -lang=java classes
//
// File is generated by gobind. Do not edit.
package java;

import go.Seq;

public abstract class Java {
    static {
        Seq.touch(); // for loading the native library
        _init();
    }
    
    private Java() {} // uninstantiable
    
    // touch is called from other bound packages to initialize this package
    public static void touch() {}
    
    private static native void _init();
    
    
    
    public static native InputStream newInputStream();
}
